pkg/containerwatcher/v2/container_watcher_collection.go

```diff
@@ -0,0 +1,138 @@
+package containerwatcher
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	containercollection "github.com/inspektor-gadget/inspektor-gadget/pkg/container-collection"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/socketenricher"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/utils/host"
+	"github.com/kubescape/go-logger"
+	"github.com/kubescape/go-logger/helpers"
+	"github.com/kubescape/node-agent/pkg/rulebindingmanager"
+	"github.com/kubescape/node-agent/pkg/utils"
+)
+
+// StartContainerCollection starts the container collection
+func (cw *ContainerWatcher) StartContainerCollection(ctx context.Context) error {
+	cw.ctx = ctx
+
+	// This is needed when not running as gadget.
+	// https://github.com/inspektor-gadget/inspektor-gadget/blob/9a797dc046f8bc1f45e85f15db7e99dd4e5cb6e5/cmd/ig/containers/containers.go#L45-L46
+	if err := host.Init(host.Config{AutoMountFilesystems: true}); err != nil {
+		return fmt.Errorf("initializing host package: %w", err)
+	}
+
+	// Start the container collection
+	containerEventFuncs := []containercollection.FuncNotify{
+		func(event containercollection.PubSubEvent) {
+			logger.L().TimedWrapper("synchronous containerCallback", 5*time.Second, func() {
+				cw.containerCallback(event)
+			})
+		},
+	}
+
+	// Initialize socket enricher for network tracers
+	if cw.cfg.EnableNetworkTracing || cw.cfg.EnableRuntimeDetection {
+		socketEnricher, err := socketenricher.NewSocketEnricher()
+		if err != nil {
+			logger.L().Error("ContainerWatcher - error creating socket enricher", helpers.Error(err))
+			return fmt.Errorf("creating socket enricher: %w", err)
+		}
+		cw.socketEnricher = socketEnricher
+	}
+
+	// Set up container callbacks
+	cw.callbacks = []containercollection.FuncNotify{
+		cw.containerCallbackAsync,
+		cw.containerProfileManager.ContainerCallback,
+		cw.objectCache.ApplicationProfileCache().ContainerCallback,
+		cw.objectCache.NetworkNeighborhoodCache().ContainerCallback,
+		cw.malwareManager.ContainerCallback,
+		cw.ruleManager.ContainerCallback,
+		cw.sbomManager.ContainerCallback,
+		cw.dnsManager.ContainerCallback,
+		cw.networkStreamClient.ContainerCallback,
+		cw.containerProcessTree.ContainerCallback,
```

Comment:
Don't you want to move the process tree callback to be before the other managers so it will reduce races?